<?xml version="1.0"?>
<launch>
  <arg name="gui" default="true" />
  <arg name="gazebo_gui" default="false" />
  <arg name="world_path"
    default="$(find jackal_kinova_gazebo)/worlds" />
  <arg name="world_name"
    default="obst_4_6by9.world" />
  <arg
    name="world" value="$(arg world_path)/$(arg world_name)" />
  <arg name="server_port"
    default="50051" />
  <arg name="x" default="0" />
  <arg name="y" default="0" />c <arg name="z"
    default="0" />
  <arg name="yaw" default="0" />
  <arg name="joystick" default="true" />
  <arg
    name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg
    if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
  <arg name="rviz_config"
    default="" />
  <arg if="$(eval rviz_config=='')" name="command_args" value="" />
  <arg
    unless="$(eval rviz_config=='')" name="command_args" value="-d $(arg rviz_config)" />
  <arg
    name="first" default="JK_01" />
  <arg name="second" default="JK_02" />

  <!-- spawn robot and bring up controllers etc. -->
  <include
    file="$(find jackal_kinova_gazebo)/launch/rl_empty_world.launch">
    <arg name="world" value="$(arg world)" />
    <arg name="gui" value="$(arg gazebo_gui)" />
    <arg name="paused" value="true" />

  </include>


  <group ns="$(arg first)">
    <!-- Load Params -->
    <rosparam command="load" file="$(find apf_robot_server)/config/jackal_params.yaml" />

    <include file="$(find jackal_kinova_gazebo)/launch/rl_spawn_jackal.launch">
      <arg name="x" value="$(arg x)" />
      <arg name="y" value="$(arg y)" />
      <arg name="z" value="$(arg z)" />
      <arg name="yaw" value="$(arg yaw)" />
      <arg name="joystick" value="$(arg joystick)" />
      <arg name="ns" value="$(arg first)" />
    </include>

    <include file="$(find jackal_kinova_moveit_config)/launch/rl_driver_launch.launch">
      <arg name="ns" value="$(arg first)" />
    </include>

    <include file="$(find jackal_kinova_gazebo)/launch/fake_localization.launch">
      <arg name="ns" value="$(arg first)" />
    </include>

    <!-- Launch robot_pose publisher -->
    <node name="robot_pose_publisher" pkg="apf_robot_server" type="robot_pose_publisher.py"
      respawn="false" output="screen">
    </node>

    <!-- Launch Goal TF Publisher -->
    <node name="goal_tf_publisher" pkg="apf_robot_server" type="pub_goal_tf.py" respawn="false"
      output="screen">
    </node>

    <!--Launch
    gRPC server -->
    <node name="robot_server" pkg="apf_robot_server" type="clpf_robot_server_with_PD.py"
      respawn="false"
      output="screen">
      <param name="server_port" value="$(arg server_port)"></param>
      <param name="real_robot" value="false"></param>
    </node>

    <node name="contact_handler" pkg="apf_robot_server" type="contact_handler.py" respawn="false"
      output="screen">
    </node>

  </group>
  <!-- <node name="rqt_tf_tree"
    pkg="rqt_tf_tree" type="rqt_tf_tree" output="screen" />
  <node name="rqt_graph"
    pkg="rqt_graph" type="rqt_graph" output="screen" />
  <node name="rqt_gui" pkg="rqt_gui"
    type="rqt_gui" output="screen">
  </node> -->

</launch>